mindsdb/interfaces/llm/llm_controller.py

```diff
@@ -0,0 +1,63 @@
+from mindsdb.interfaces.storage import db
+from mindsdb.utilities.config import Config
+
+
+class LLMDataController:
+    '''Handles CRUD operations at the database level for LLM Data'''
+
+    def __init__(self) -> None:
+        '''Initializes the LLMDataController with configuration settings.'''
+        self.config = Config()
+
+    def add_llm_data(self, input_data: str, output_data: str, model_id: int) -> db.LLMData:
+        '''
+        Adds LLM input and output data to the database.
+        Parameters:
+            input_data (str): The input to the LLM.
+            output_data (str): The output from the LLM.
+        Returns:
+            LLMData: The created LLMData object.
+        '''
+        # TODO: check the hash to avoid adding duplicate records (SQLAlchemy-based)

```

Comment:
This is fairly important to reason about. I think we _do_ want duplicate records at different timestamps, as they could indicate incorrect tool usage, for example. If you agree, we can remove this comment and keep implementation as-is, because the key in this table is `id`, so duplicate records are fine.

Reply:
yes makes sense